src/main/java/com/mongodb/kafka/connect/source/MongoSourceTask.java

```diff
@@ -244,7 +244,7 @@ public synchronized void stop() {
             if (resumeToken != null) {
                 if (e.getErrorCode() == 260) {
                     invalidatedCursor = true;
-                    return tryCreateCursor(sourceConfig, mongoClient, resumeToken);
+                    return tryCreateCursor(sourceConfig, mongoClient, null);
```

Comment:
Made the resumeToken explicitly null - this path occurs when a resumeAfter is used with an invalidate token and in that case supportsStartAfter would be false - but as we're invalidating here it makes sense to explicitly exclude the resumeToken when trying again to create the cursor.